src/helpers/customShuffle.js

```diff
@@ -0,0 +1,480 @@
+            numOptions: 0
+        }
+        var token = selectFromType("Artificial", terraformsByLandType)
+        token.location = {x: previousTile.location.x, y:previousTile.location.y}
+        assignedTiles.push(token)
+        delete token.pool_link
+        delete token.image
+    }
+
+    //iterate for all the terraforms in each sector
+    for (let i = 0; i < sectorTerraforms.length; i++){
+        //only select less common land types first
+        if (landTypeCounts["Artificial"] !== 0 || landTypeCounts["Badlands"] !== 0 || landTypeCounts["Desert"] !== 0 || landTypeCounts["Fungal"] !== 0 || landTypeCounts["Tundra"] !== 0 || landTypeCounts["Volcanic"] !== 0 || landTypeCounts["Wasteland"] || landTypeCounts["Rocky" !== 0]){
+            var chooseRare = true;
+        }else{chooseRare = false}
+        
+        //select tile and extract relevant info
+        var selectedTile = selectTile(possibleTiles, previousTile, landTypeCounts[previousTile.outType] !== 0)
+        var selectedType = selectedTile.outType
+        var selectedX = selectedTile.location.x
+        var selectedY = selectedTile.location.y
+        var selectedIdx = xyToIdx(selectedX, selectedY, gridDim)
+        
+        //choose token out of available of that type
+        var token = selectFromType(selectedType, terraformsByLandType)
+        //sanitize token
+        delete token.pool_link
+        delete token.image
+
+        token.location = { 
+            x:selectedX, 
+            y:selectedY}
+
+        if ( i == 0){
+            assignedTiles = [token]
+        }else{
+            assignedTiles.push(token)
+        }
+
+        assignedTilesOrdered[selectedIdx] = token
+
+        //remove land from land type
+        terraformsByLandType[selectedType] = terraformsByLandType[selectedType].filter(land => land.token_name !== token.token_name)
+
+        //remove ability to select land again
+        possibleTiles[selectedIdx].possible_types = []
+
+        propagateConstraints(token, possibleTiles, gridDim, sectorTerraforms.length)
+
+        landTypeCounts = updateLandTypeCounts(terraformsByLandType)
+
+        //Checks for land counts to hit 0 and updates all possible tiles without that type
+        for (var land in landTypeCounts){
+            if (landTypeCounts[land] == 0) {
+                possibleTiles.forEach( tile => {
+                    tile.possible_types = tile.possible_types.filter(type => type !== land)
+                    tile.numOptions = tile.possible_types.length
+                })           
+            }
+        } 
+ 
+        previousTile = selectedTile;
+    }
+
+    sequentialTerraforms[sectors[t]] = assignedTilesOrdered
+    shuffledTerraforms[sectors[t]] = assignedTiles
+    console.log("Finished Shuffling", sectors[t], "Sector")
+}
+
+export var orderedTerraforms = sequentialTerraforms
+
+
+// function to select a tile, add it to assigned tiles, and remove it from possible tiles
```

Comment:
comment is no longer accurate. The function only selects and returns a terraform (tile)